### PR TITLE
feat(config): validate file config on repo runs

### LINKED ISSUE #25274: Validate config.js and _WARN_ for any config errors
### Describe the proposed change(s).

Global config (e.g. config.js, but also including `RENOVATE_CONFIG`) should be validated and a log warning issued if they contain invalid values.

This warning should *not* show up in _repos_ (e.g. in dependency dashboard).

We will upgrade this severity to ERROR and then FATAL in future major releases

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Migrate and validate file config and `RENOVATE_CONFIG` during runtime. The errors/warnings from the validation will be logged as warnings only, and will not appear on dependency dashboard.

## Context
Closes: https://github.com/renovatebot/renovate/issues/25274 and https://github.com/renovatebot/renovate/issues/23868

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/config/parse/file.ts
Comment (RahulGautamSingh) [inline lib/workers/global/config/parse/file.ts:97]: Should I use the name of the configuration here? ie. file config or bot config.
It might confuse users if we don't mention which config the errors are related to.

**Comment by RahulGautamSingh** [comment]
Test Runs: https://github.com/RahulGautamSingh-testing/repro-26379

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by RahulGautamSingh** [comment]
Directly using `validateConfig` could be a breaking change, since this file contains global config. And currently this fn doesn't allow
global options of type `Object`.

**Comment by rarkins** [comment]
Please describe in more detail what you mean

**Comment by RahulGautamSingh** [comment]
The `validateConfig` function which we use for validation and which I have added in the `env.ts`, `file.ts` files to validate `RENOVATE_CONFIG` and `file-config`, has one issue.

The issue is that will throw an error if any global options of type `Object` are present in the config. It might seem confusing as why only the global options of `Object` type are affected. It is because the validation is broken and a fix PR is open for that.

The `RENOVATE_CONFIG` might not be affected by this issue because it generally contains only non-global options but if we validate the `file-config` using this function then users will get warnings for the global options present inside it. 

My bad. It's not breaking since the repo will be processed, just not correct.

**Comment by rarkins** [comment]
@RahulGautamSingh #25218 is classified as a feature, not a fix. If there's a fix needed, please separate it first

**Comment by RahulGautamSingh** [comment]
> @RahulGautamSingh #25218 is classified as a feature, not a fix. If there's a fix needed, please separate it first

My bad, I meant to link this PR: https://github.com/renovatebot/renovate/pull/24561 .. In this one we separate the validate for global and non-global options and in https://github.com/renovatebot/renovate/pull/25218 we add validation for global config.

So this PR needs to wait for both the linked PRs.

**Review by viceice — COMMENTED** [review]
please mark as draft until dependent PR's are merged. 🤗

**Comment by RahulGautamSingh** [comment]
Blocked By: https://github.com/renovatebot/renovate/pull/25218

File: lib/workers/global/config/parse/env.ts
Comment (viceice) [inline lib/workers/global/config/parse/env.ts:127]: Extract to shared function (`lib/workers/global/config/parse/util.ts`), same code is used on file.ts

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/global/config/parse/file.ts
Comment (viceice) [inline lib/workers/global/config/parse/file.ts:85]:
```suggestion
  return migrateAndValidateConfig(config, configFile);
```
no need to wait here.

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**
